src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -125,13 +131,65 @@ def _DiracToArcID(self, diracJobID):
         self.log.warn("Identifier already in REST format?", diracJobID)
         return diracJobID
 
+    #############################################################################
+
     def _urlJoin(self, command):
         """Add the command to the base URL.
 
         :param str command: command to execute
         """
         return os.path.join(self.base_url, command)
 
+    def _request(self, method, query, params=None, data=None, headers=None, timeout=None):
+        """Perform a request and properly handle the results/exceptions.
+
+        :param str method: "post", "get", "put"
+        :param str query: query to submit
+        :param dict/str params: parameters of the query
+        :param dict headers: headers of the query
+        :param int timeout: timeout value
+        :return: response
+        """
+        if not headers:
+            headers = self.headers
+        if not timeout:
+            timeout = self.timeout
+
+        if method.upper() not in ["GET", "POST", "PUT"]:
+            return S_ERROR("The request method is unknown")
+
+        try:
+            response = self.session.request(
+                method,
+                query,
+                headers=headers,
+                params=params,
+                data=data,
+                timeout=timeout,
+            )
+            if not response.ok:
+                return S_ERROR("Response: %s - %s" % (response.status_code, response.reason))
+            return S_OK(response)
+        except requests.Timeout as e:
+            return S_ERROR("Request timed out, consider increasing the Timeout value: %s" % e)
+        except requests.ConnectionError as e:
+            return S_ERROR("Connection failed, consider checking the state of the CE: %s" % e)
+        except requests.RequestException as e:
+            return S_ERROR("Request exception: %s" % e)
+
+    def _checkSession(self):
```

Comment:
Well, currently the decorator is not used within `AREXCE` since it is not fully adapted to it.

> which looks like a better solution

Why is it a better solution?

While I see the advantage of generic decorators used in many different classes, I don't fully understand the need for a custom decorator used in a single class: `ARC` (`ARC6` inherits from `ARC`).
The decorator proposed in https://github.com/DIRACGrid/DIRAC/pull/6803 is very specific and cannot be used in `AREX`.

The `_checkSession()` could actually be renamed, added in `ARC` and be overridden in `ARC6` and `AREX`.

Reply:
> The decorator proposed in #6803 is very specific and cannot be used in `AREX`.

It cannot? They look rather similar to me. The decorator is simply there to avoid code repetition.

Reply:
> It cannot? They look rather similar to me.

It cannot because of this line: https://github.com/DIRACGrid/DIRAC/pull/6803/files#diff-399d757a2e7244ab09d3f3f3348fe6dab3bd0c1f68af60372a66b781e9795ac5R102

`self.usercfg` does not exist in `AREX`.
We could move this line, but then we would need to call it in every method using a proxy. 

> The decorator is simply there to avoid code repetition.

Where would we reuse this decorator?
IMHO, in this specific context, the content of the decorator is too much specific, and a method seems more adapted.

Reply:
The line above can be executed only if self.usercfg attribute exists. However, I do not think it is actually worth doing. I agree with Alexandre that ARC(6) and AREX are rather different in details despite the fact that they do almost the same thing.